kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ReplicaManager.java

```diff
@@ -111,6 +111,7 @@ public CompletableFuture<Map<TopicPartition, ProduceResponse.PartitionResponse>>
                         .thenAccept(offset -> addPartitionResponse.accept(topicPartition,
                                 new ProduceResponse.PartitionResponse(Errors.NONE, offset, -1L, -1L)))
                         .exceptionally(ex -> {
+                            log.error("Unexpected error when append records.", ex);
```

Comment:
It's not an unexpected error unless `Errors.forException(ex.getCause())` returns `UNKNOWN_SERVER_ERROR`. This error is usually `NOT_LEADER_FOR_PARTITION` that trigger the retry of clients, in this case, we should not print the full stack.